File: week2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int letterCount = 0, wordCount = 0, sentenceCount = 0;
    
    // Get input from user
    string inputString = get_string("Enter: ");
    
    // Special chars specifying end of sentence
    const char *endChars = "?!.";
    char *c = inputString;
    
    // Loop across each character 
    while (*c)
    {
        // Check letter
        if (isalpha(*c))
        {
            letterCount++;
        }
        // Check space
        if (isspace(*c))
        {
            // For first space encountered increment wordCount
            if (!wordCount)
            {
                wordCount++;
            }
            wordCount++;
        }
        // Check if character is a sentenace endpoint
        if (strchr(endChars, *c))
        {
            sentenceCount++;
        }
        c++;
    }  

    // Coleman-Liau formula
    float L = 100 * letterCount / (float) wordCount;
    float S = 100 * sentenceCount / (float) wordCount;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %i\n", index);
    }
}

File: week6/readability.py
from cs50 import get_string
from sys import exit

letterCount = wordCount = sentenceCount = 0

inputString = get_string("Enter: ")
# Characters signifying end of sentence
endChars = "?!."

# Loop across the string
for c in inputString:
    # Check letter
    if c.isalpha():
        letterCount += 1
    # Check space
    if c == " ":
        # For first space encountered increment wordCount
        if (not wordCount):
            wordCount += 1
        wordCount += 1
    # Check if character is a sentenace endpoint
    if c in endChars:
        sentenceCount += 1

# Coleman-Liau formula
L = 100 * letterCount / wordCount
S = 100 * sentenceCount / wordCount
index = round(0.0588 * L - 0.296 * S - 15.8)

if (index < 1):
    print("Before Grade 1\n")
elif (index > 16):
    print("Grade 16+\n")
else:
    print(f"Grade {index}\n") 
